doc/build-helpers/testers.chapter.md

```diff
@@ -120,9 +120,10 @@ It has two modes:
 
 Checks that the output from running a command contains the specified version string in it as a whole word.
 
-Although simplistic, this test assures that the main program can run.
-While there's no substitute for a real test case, it does catch dynamic linking errors and such.
-It also provides some protection against accidentally building the wrong version, for example when using an "old" hash in a fixed-output derivation.
+NOTE: In most cases, [`versionCheckHook`](#versioncheckhook) should be preferred, but this function is provided and documented here anyway. The motivation for adding either tests would be:
```

Comment:
For content addressed derivations, a passthru test would be preferable because it tests the executable after rewriting instead of before, and rewriting may break the executable.

Reply:
> For content addressed derivations, a passthru test would be preferable because it tests the executable after rewriting instead of before, and rewriting may break the executable.

Hmm interesting.. Perhaps we can check inside the hook whether `__contentAddressed == true` and output a message or something like that? We barely have enough content addressed derivations at the moment in Nixpkgs so I don't know how would we test the need for this check...